src/CaptivePortal.cpp

```diff
@@ -117,12 +176,78 @@ bool CaptivePortal::BroadcastMessageBIN(const std::uint8_t* data, std::size_t le
 }
 
 void handleWebSocketClientConnected(std::uint8_t socketId) {
-  ESP_LOGD(
-    TAG, "WebSocket client #%u connected from %s", socketId, s_webServices->socketServer.remoteIP(socketId).toString().c_str());
+  ESP_LOGD(TAG, "WebSocket client #%u connected from %s", socketId, s_webServices->socketServer.remoteIP(socketId).toString().c_str());
+
+  StaticJsonDocument<24> doc;
+  doc["type"] = "poggies";
+  CaptivePortal::SendMessageJSON(socketId, doc);
 }
 void handleWebSocketClientDisconnected(std::uint8_t socketId) {
   ESP_LOGD(TAG, "WebSocket client #%u disconnected", socketId);
 }
+void handleWebSocketClientWiFiScanMessage(const StaticJsonDocument<256>& doc) {
+  bool run = doc["run"];
+  if (run) {
+    WiFiScanManager::StartScan();
+  } else {
+    WiFiScanManager::CancelScan();
+  }
+}
+void handleWebSocketClientWiFiAuthenticateMessage(const StaticJsonDocument<256>& doc) {
+  String bssidStr = doc["bssid"];
+  if (bssidStr.isEmpty()) {
+    ESP_LOGE(TAG, "WiFi BSSID is missing");
+    return;
+  }
+  if (bssidStr.length() != 17) {
+    ESP_LOGE(TAG, "WiFi BSSID is invalid");
+    return;
+  }
+
+  String password = doc["password"];
+
+  // Convert BSSID to byte array
+  std::uint8_t bssid[6];
+  if (sscanf(bssidStr.c_str(), "%02X:%02X:%02X:%02X:%02X:%02X", bssid + 0, bssid + 1, bssid + 2, bssid + 3, bssid + 4, bssid + 5) != 6) {
+    ESP_LOGE(TAG, "WiFi BSSID is invalid");
+    return;
+  }
```

Comment:
The comment tells me about a conversion, but I don't understand the leap between [that] and `"Wifi BSSID is invalid"`. Add a comment about what it is you're checking/trying to achieve?

Reply:
Added a comment:

Uses sscanf to parse the max-style hex format, e.g. "AA:BB:CC:DD:EE:FF" where each pair is a byte, and %02X means to parse 2 characters as a hex byte
We check the return value to ensure that we parsed all 6 arguments (6 pairs of hex bytes, or 6 bytes)